### PR TITLE
*: Fix  autocommit 

### PR DESCRIPTION
**Pull request by zimulala**
fix https://github.com/go-xorm/tidb/issues/2

### CODE REVIEW

File: bootstrap.go [hunk 1]
```diff
@@ -156,7 +157,14 @@ func checkBootstrappedVar(s Session) (bool, error) {
 	if err != nil || row == nil {
 		return false, errors.Trace(err)
 	}
-	return row.Data[0].(string) == bootstrappedVarTrue, nil
+
+	isBootstrapped := row.Data[0].(string) == bootstrappedVarTrue
+	if isBootstrapped {
+		// Make sure that doesn't affect the next operations.
+		s.FinishTxn(false)
```
Comment (siddontang) [inline bootstrap.go]: check error here.

File: bootstrap.go [hunk 2]
```diff
@@ -156,7 +157,14 @@ func checkBootstrappedVar(s Session) (bool, error) {
 	if err != nil || row == nil {
 		return false, errors.Trace(err)
 	}
-	return row.Data[0].(string) == bootstrappedVarTrue, nil
+
+	isBootstrapped := row.Data[0].(string) == bootstrappedVarTrue
+	if isBootstrapped {
+		// Make sure that doesn't affect the next operations.
```
Comment (shenli) [inline bootstrap.go]: "the next operation" or "the following operations"

File: session_test.go [hunk 3]
```diff
@@ -734,9 +735,31 @@ func (s *testSessionSuite) TestBootstrap(c *C) {
...
+	se.Close()
+	se, err = CreateSession(store)
+	c.Assert(err, IsNil)
+	mustExecSQL(c, se, "USE test;")
+	mustExecSQL(c, se, "insert t values (?)", 3)
+	se, err = CreateSession(store)
+	c.Assert(err, IsNil)
+	mustExecSQL(c, se, "USE test;")
+	r = mustExecSQL(c, se, "select * from t")
+	c.Assert(r, NotNil)
```
Comment (shenli) [inline session_test.go]: check the values in r?
Even if table t is empty, r will not be nil.

Comment (qiuyesuifeng) [inline session_test.go]: Seems line 760 has checked r values?
But i think it is needless to check whether r is null, for if r is null, line 758 will panic.

Comment (shenli) [inline session_test.go]: I got it.

File: bootstrap.go (hunk 2 above)
Comment (zimulala) [inline bootstrap.go]: "the following operations" is better.

File: bootstrap.go (hunk 1 above)
Comment (zimulala) [inline bootstrap.go]: got it.

**Comment by qiuyesuifeng** [comment]
LGTM

**Comment by shenli** [comment]
LGTM

**Pull request merged**